sootup.callgraph/src/main/java/sootup/callgraph/AbstractCallGraphAlgorithm.java

```diff
@@ -191,7 +299,23 @@ final <T extends Method> T findMethodInHierarchy(
   }
 
   /**
-   * This method enables optional post processing of a method in the call graph algorithm
+   * This method enables optional pre-processing of a method in the call graph algorithm
+   *
+   * @param view view
+   * @param sourceMethod the processed method
+   * @param workList the current worklist that might be extended
+   * @param cg the current cg that might be extended
+   */
+  public void preProcessingMethod(
+      View<? extends SootClass<?>> view,
+      MethodSignature sourceMethod,
+      @Nonnull Deque<MethodSignature> workList,
+      @Nonnull MutableCallGraph cg) {
+    // This is only implemented if it is needed in the call graph algorithm
+  }
```

Comment:
then you have to create this dummy method for all other algorithm which are not RTA
But yeah we can do that, so that you are aware of this steps in the algorithm

Reply:
I think it's better to make it explicit.